mlir/lib/Analysis/Presburger/Simplex.cpp

```diff
@@ -104,8 +104,9 @@ Simplex::Unknown &SimplexBase::unknownFromRow(unsigned row) {
 
 unsigned SimplexBase::addZeroRow(bool makeRestricted) {
   // Resize the tableau to accommodate the extra row.
+  unsigned oldNumRows = getNumRows();
   unsigned newRow = tableau.appendExtraRow();
-  assert(getNumRows() == getNumRows() && "Inconsistent tableau size");
+  assert(getNumRows() == oldNumRows + 1 && "Inconsistent tableau size");
```

Comment:
you can just delete the assert. it's a vestige of an earlier time when there was an extra variable tracking the number of rows that had to be kept in sync with the Matrix internal count.

Reply:
Thanks for catching this!